### ISSUE TITLE
Missing {pres} message on p2p topic creation (reopen + one related issue)

### ISSUE THREAD

**Issue by freele** [tinode/chat#583]
Re-opening the issue https://github.com/tinode/chat/issues/575

recap
This issue originally was considered as one of the cases described in https://github.com/tinode/chat/issues/574 (Problem 2)
Which is marked as fixed in 3193ee2d884abc7676feb800faca5c2fa9419924.

Anyways, I am cherry-picking this commit to the latest release, and the problem is still present.
I can not create a p2p topic for two users who do not know about each other beforehand.

Besides, after this commit connection started to drop on this test (out of context, but it works without this commit). I presume, it is related, so I do not create a separate issue for this one:
```ts
    it('user (2) upgrades WANT via set.sub', async () => {
      const [, userTwo] = users
      const { stream } = userTwo

      const [ctrl] = await sendAndWait<pbx.IClientSet, pbx.IServerCtrl>(stream, 'set', {
        topic: grpId,
        query: {
          sub: {
            mode: 'JRWPAS'
          }
        }
      })

      assert.equal(ctrl.code, 200)
      await wait(300)
    })
```
  
Now I have a connection drop here, here is the log
```
Error: 14 UNAVAILABLE: Connection dropped
    at Object.callErrorFromStatus (/src/node_modules/@grpc/grpc-js/src/call.ts:81:24)
    at Object.onReceiveStatus (/src/node_modules/@grpc/grpc-js/src/client.ts:666:32)
    at Object.onReceiveStatus (/src/node_modules/@grpc/grpc-js/src/client-interceptors.ts:397:48)
    at Http2CallStream.outputStatus (/src/node_modules/@grpc/grpc-js/src/call-stream.ts:230:22)
    at Http2CallStream.maybeOutputStatus (/src/node_modules/@grpc/grpc-js/src/call-stream.ts:280:14)
    at Http2CallStream.endCall (/src/node_modules/@grpc/grpc-js/src/call-stream.ts:263:12)
    at Http2CallStream.disconnectListener (/src/node_modules/@grpc/grpc-js/src/call-stream.ts:215:12)
    at Socket.<anonymous> (/src/node_modules/@grpc/grpc-js/src/subchannel.ts:524:13)
    at Object.onceWrapper (events.js:422:26)
    at Socket.emit (events.js:327:22)
    at TCP.<anonymous> (net.js:674:12)
```
Let me know, if you need more logs, I have not done explicit investigation yet, just tried several times with/without the fix mentioned above.

**Comment by or-else** [comment]
> I can not create a p2p topic for two users who do not know about each other beforehand.

Please show exact steps to reproduce using stock tools.

> Now I have a connection drop here, here is the log

Please show the server-side log.

In general, please do use bug templates. Please file separate issues for separate problems.

**Comment by SatoshiNakamotogmx** [comment]
What if they both were guests and admin?

**Comment by freele** [comment]
@or-else 
Could you just edit your fix mentioned above (https://github.com/tinode/chat/commit/3193ee2d884abc7676feb800faca5c2fa9419924) to something like
```go
if (pkt.Sub != nil && pkt.Sub.Newsub) || oldWant != userData.modeWant || oldGiven != userData.modeGiven {
```
I suppose that should fix everything. At least it works on my local tests. If it does not, I will dig deeper and come back with a clear reproduction scenario.

**Comment by or-else** [comment]
Yes, you are right. Indeed it may crash here. Fixed:
https://github.com/tinode/chat/commit/ffac63163b6ccf26f37edc3f3e71fb9115f2c01b

**Comment by or-else** [comment]
Released in v0.16.9-rc2
